library/components/TTable.vue

```diff
@@ -0,0 +1,1018 @@
+          header: _.toString(columnString),
+          property: columnString,
+        };
+      });
+
+      // exclude the grouping column
+      if (this.group_by_column) {
+        cols = cols.filter((col) => col.property !== this.group_by_column);
+      }
+
+      if (this.columnConfigs) {
+        this.columnConfigs.forEach((columnConfig) => {
+          const defaultConfigIndex = cols.findIndex((dc) => {
+            return dc.property === columnConfig.property;
+          });
+
+          if (defaultConfigIndex !== -1) {
+            const defaultConfig = cols[defaultConfigIndex];
+            if (!columnConfig.header) {
+              columnConfig.header = defaultConfig.header;
+            }
+            if (
+              columnConfig &&
+              columnConfig.sort &&
+              columnConfig.sort.priority
+            ) {
+              columnConfig.sort.priority = Number(columnConfig.sort.priority);
+            }
+            cols.splice(defaultConfigIndex, 1, columnConfig);
+          } else {
+            cols.push(columnConfig);
+          }
+        });
+      }
+      if (this.can_sort) {
+        // Note: the "+" in "+col.sort.priority" converts strings to numbers
+        const sortPriorities = cols
+          .map((col) => (col.sort ? +col.sort.priority : null))
+          .filter((priority) => priority !== null)
+          .sort();
+
+        let maxSortPriority =
+          sortPriorities.length > 0 ? sortPriorities.pop() : 0;
+
+        cols.forEach((col) => {
+          if (!col.sort) {
+            maxSortPriority += 1;
+            col.sort = {
+              direction: this.sort_direction,
+              priority: maxSortPriority,
+            };
+          }
+        });
+      }
+
+      // remove sorting on calculated columns for now.
+      // TODO: figure out if there is a way to get the computed value in each cell slot so that dynamic columns can be sorted
+      const baseColumns = Object.keys(this.rows[0]);
+      cols.forEach((col) => {
+        if (!baseColumns.includes(col.property)) {
+          col.sort = null;
+        }
+      });
+
+      return cols;
+    },
+    layerRows() {
+      if (this.rows) return this.rows;
+      return [];
+    },
+  },
+  watch: {
+    internalSelectedItem() {
+      this.$emit("update:selectedItem", this.internalSelectedItem);
+      this.$emit("selectedItemChanged", this.internalSelectedItem);
+    },
+    internalSelectedItems() {
+      this.$emit("update:selectedItems", this.internalSelectedItems);
+      this.$emit("selectedItemsChanged", this.internalSelectedItems);
+    },
+    layerRows: {
+      deep: true,
+      handler() {
+        this.init();
+      },
+    },
+  },
+  methods: {
+    updateSearchTerm(newSearchTerm) {
+      this.searchTerm = newSearchTerm;
+    },
+    setColumnConfig(columnConfig) {
+      this.columnConfigs = this.columnConfigs.concat(columnConfig);
+      this.init();
+      this.isDataAvailable = true;
+    },
+    init() {
```

Comment:
A lot is happening in this method and the ones below, wish I could help for those.
Splitting those into smaller functions may be quite helpful already. 👌🏼

Reply:
Believe it or not, those methods are the end result of a major refactor! They do handle a lot, I'll take another look and see if I can't simplify/extract some things. At least these days they all only handle one conceptual thing and they do so independent of the others. The original code was straight up bad.